gpt_index/indices/query/embedding_utils.py

```diff
@@ -31,3 +32,32 @@ def get_top_k_embeddings(
     result_ids = [n for _, n in result_tups]
 
     return result_similarities, result_ids
+
+
+class SimilarityTracker:
+    """Helper class to manage node similarities during lifecycle of a single query."""
```

Comment:
Is the goal of this class mostly just to track similarities? 

I have an alternative proposal - there's already an `extra_info` field in Node. If similarity can be computed, then just update extra_info with `{"similarity": value}`. This way:
- You avoid needing to specify node_kwargs in _get_nodes_for_response
- You don't need to define a SimilarityTracker class

Reply:
I briefly considered that, and would love to avoid the kwargs stuff, but isn't it kind of weird to intermingle similarity with metadata about the document that's passed down to the node? 

Moreover, the similarity score is specific to each *query* and should change each time, whereas my understanding is that the nodes exist at a higher level and their contents (including extra_info) don't change.

Reply:
Yeah you're right. What are your thoughts on the updated proposals?

Reply:
I'm good with (1). Actually, (2) was the first thing I tried but it felt really ugly especially since most of the child implementations of _get_nodes_for_response don't actually have similarities.

Reply:
^ edited